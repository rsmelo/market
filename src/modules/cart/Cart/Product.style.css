.image {
  width: 100px;
  height: 120px;
}

.productCell {
  vertical-align: middle;
  padding: 0.4em;
}

.buttonCell {
  text-align: right;
}

.smallCell {
  width: 10%;
}

.largeCell {
  width: 40%;
}
