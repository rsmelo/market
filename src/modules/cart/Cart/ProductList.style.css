.productList {
  width: 100%;
  border-collapse: collapse;
  border-spacing: 0;
  border: 1px solid #e5e5e5;
}

.productList tr {
  border: 1px solid #e5e5e5;
}

.header {
  border: 1px solid transparent;
}

.header > tr > th {
  border: none;
  padding: 0 5px;
  vertical-align: middle;
  line-height: 1.7;
  text-align: left;
}

.header,
.footer {
  background: var(--color-10-grey);
  color: var(--color-dark-grey);
}

.totalLabel {
  font-weight: bold;
  text-align: right;
  font-size: 16px;
  line-height: 1.7;
  padding: 0 5px;
}
