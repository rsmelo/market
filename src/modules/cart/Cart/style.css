.container {
  margin-top: 4em;
}

.cartContainer {
  padding: 0 2em;
  margin-top: 1em;
}
