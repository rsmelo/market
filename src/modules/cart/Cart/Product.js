import React from 'react'
import PropTypes from 'prop-types'

import Button from '../../../components/Button'
import style from './Product.style.css'

const Product = ({
  id,
  image,
  name,
  price,
  seller,
  onRemove,
}) => (
  <tr>
    <td>
      <img
        src={image}
        alt={name}
        className={style.image}
      />
    </td>
    <td className={style.productCell}>{name}</td>
    <td className={style.productCell}>R$ {price}</td>
    <td className={style.productCell}>
      <Button onClick={() => onRemove(id, seller.id)}>
        Remover do carrinho
      </Button>
    </td>
  </tr>
)

Product.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  image: PropTypes.string.isRequired,
  seller: PropTypes.shape({
    id: PropTypes.string.isRequired,
  }).isRequired,
  onRemove: PropTypes.func.isRequired,
}
export default Product
