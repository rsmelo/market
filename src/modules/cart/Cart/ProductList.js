import React from 'react'
import PropTypes from 'prop-types'

import Product from './Product'
import style from './ProductList.style.css'
import calculateTotal from '../../../utils/calculateTotal'


const ProductList = ({ products, removeProduct }) => (
  <table className={style.productList}>
    <thead className={style.header}>
      <tr>
        <th colSpan="2">Produto(s)</th>
        <th >Pre√ßo</th>
        <th />
      </tr>
    </thead>
    <tbody>
      {products.map(product => (
        <Product
          key={product.id}
          {...product}
          onRemove={removeProduct}
        />
      ))
      }
    </tbody>
    <tfoot className={style.footer}>
      <tr>
        <td colSpan="2" />
        <td />
        <td className={style.totalLabel}>
          <span>Total</span> R$ {calculateTotal(products)}
        </td>
      </tr>
    </tfoot>
  </table>
)

ProductList.propTypes = {
  products: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  removeProduct: PropTypes.func.isRequired,
}

export default ProductList
