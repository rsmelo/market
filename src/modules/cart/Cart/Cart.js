import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import ProductList from './ProductList'
import ButtonLink from '../../../components/ButtonLink'
import style from './style.css'

class Cart extends PureComponent {
  constructor (props) {
    super(props)
    this.renderCarts = this.renderCarts.bind(this)
  }

  renderCarts () {
    const { carts, removeProduct } = this.props
    return carts.map(({ seller, products }) => (
      <div key={seller.id}>
        <h2 className={style.seller}>Vendido por {seller.name}</h2>
        <ProductList
          products={products}
          removeProduct={removeProduct}
        />
        <div className={style.submitButtonContainer}>
          <ButtonLink to={`/checkout/${seller.id}`}>
            Finalizar pedido
          </ButtonLink>
        </div>
      </div>
    ))
  }

  render () {
    const { carts } = this.props
    return (
      <div className={style.outsideBox}>
        <div className={style.container}>
          <div className={style.backButtonContainer}>
            <ButtonLink to="/" appearance="outline">
              Continuar comprando
            </ButtonLink>
          </div>
          {carts.length ?
            this.renderCarts() :
            (<p>Você não possui nenhum produto no carrinho</p>)
          }
        </div>
      </div>
    )
  }
}

Cart.propTypes = {
  carts: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  removeProduct: PropTypes.func.isRequired,
}

export default Cart
