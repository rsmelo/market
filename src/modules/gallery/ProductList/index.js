import { connect } from 'react-redux'

import ProductList from './ProductList'

import { getProducts } from '../actions'
import { getGalleryProducts } from '../../combinedReducers'

const mapStateToProps = state => ({
  products: getGalleryProducts(state),
})

const mapDispatchToProps = {
  getProducts,
}


export default connect(mapStateToProps, mapDispatchToProps)(ProductList)
