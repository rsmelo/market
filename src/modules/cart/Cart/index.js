import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'

import Cart from './Cart'

import { getCartList } from '../../combinedReducers'
import { removeProduct } from '../actions'

const mapStateToProps = state => ({
  carts: getCartList(state),
})

const mapDispatchToProps = {
  removeProduct,
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Cart))
