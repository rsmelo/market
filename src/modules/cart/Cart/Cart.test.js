import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import Cart from './Cart'
import ProductList from './ProductList'

jest.mock('react-router-dom', () => ({ Link: 'Link' }))

describe('Cart', () => {
  it('should render correctly', () => {
    const seller = {
      id: 're_cjabsxlgq01t5oq6f3ix79dsn',
      name: 'João',
    }
    const carts = [{
      seller,
      products: [
        {
          id: 'dd5ca2b6-b8ae-4627-95f3-7d49613a0df5',
          name: 'Abominável',
          price: 80,
          image: 'https://images-na.ssl-images-amazon.com/images/I/A1eI8aYKX%2BL._SL1500_.jpg',
          description: 'description',
          category: 'action figures',
          seller,
        },
      ],
    }]
    const removeProduct = jest.fn()
    const tree = renderer
      .create(<Cart carts={carts} removeProduct={removeProduct} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should have a ProductList to each item in carts property', () => {
    const removeProduct = jest.fn()
    const seller1 = {
      id: 're_cjabsxlgq01t5oq6f3ix79dsn',
      name: 'João',
    }
    const seller2 = {
      id: 're_cjabsxlgq01t5oq6f3ix79abc',
      name: 'Pedro',
    }
    const carts = [
      {
        seller: seller1,
        products: [
          {
            id: 'dd5ca2b6-b8ae-4627-95f3-7d49613a0df5',
            name: 'Abominável',
            price: 80,
            image: 'https://images-na.ssl-images-amazon.com/images/I/A1eI8aYKX%2BL._SL1500_.jpg',
            description: 'description',
            category: 'action figures',
            seller: seller1,
          },
        ],
      },
      {
        seller: seller2,
        products: [
          {
            id: 'dd5ca2b6-b8ae-4627-95f3-7d49613a1234',
            name: 'prod',
            price: 80,
            image: 'https://images.com/1.jpg',
            description: 'description',
            category: 'action figures',
            seller: seller2,
          },
        ],
      },
    ]

    const component = shallow(<Cart removeProduct={removeProduct} carts={carts} />)
    const productsList = component.find(ProductList)
    expect(productsList).toHaveLength(2)
  })

  it('should show the message "Você não possui nenhum produto no carrinho" when carts length equal zero', () => {
    const removeProduct = jest.fn()

    const component = shallow(<Cart removeProduct={removeProduct} carts={[]} />)
    const productsList = component.find('p')
    expect(productsList.text()).toBe('Você não possui nenhum produto no carrinho')
  })
})
