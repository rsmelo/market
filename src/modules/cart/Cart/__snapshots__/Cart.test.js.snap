// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart should render correctly 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <Link
        className=""
        to="/"
      >
        Continuar comprando
      </Link>
    </div>
    <div>
      <h4
        className={undefined}
      >
        Vendido por 
        João
      </h4>
      <table
        className={undefined}
      >
        <thead
          className={undefined}
        >
          <tr>
            <th
              colSpan="2"
            >
              Produto(s)
            </th>
            <th>
              Preço
            </th>
            <th />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              className=""
            >
              <Link
                to="/product/dd5ca2b6-b8ae-4627-95f3-7d49613a0df5"
              >
                <img
                  alt="Abominável"
                  className={undefined}
                  src="https://images-na.ssl-images-amazon.com/images/I/A1eI8aYKX%2BL._SL1500_.jpg"
                />
              </Link>
            </td>
            <td
              className=""
            >
              Abominável
            </td>
            <td
              className=""
            >
              R$ 
              80
            </td>
            <td
              className=""
            >
              <button
                className=""
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Remover do carrinho
              </button>
            </td>
          </tr>
        </tbody>
        <tfoot
          className={undefined}
        >
          <tr>
            <td
              colSpan="2"
            />
            <td />
            <td
              className={undefined}
            >
              <span>
                Total
              </span>
               R$ 
              80
            </td>
          </tr>
        </tfoot>
      </table>
      <div
        className={undefined}
      >
        <Link
          className=""
          to="/checkout/re_cjabsxlgq01t5oq6f3ix79dsn"
        >
          Finalizar pedido
        </Link>
      </div>
    </div>
  </div>
</div>
`;
